crates/loro-internal/src/event.rs

```diff
@@ -112,13 +113,53 @@ impl<'a> InternalDocDiff<'a> {
 
 pub type Path = SmallVec<[Index; 4]>;
 
-#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, enum_as_inner::EnumAsInner)]
 pub enum Index {
     Key(InternalString),
     Seq(usize),
     Node(TreeID),
 }
 
+impl std::fmt::Display for Index {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Index::Key(key) => write!(f, "{}", key),
+            Index::Seq(s) => write!(f, "{}", s),
+            Index::Node(id) => write!(f, "{}@{}", id.peer, id.counter),
+        }
+    }
+}
+
+impl TryFrom<&str> for Index {
+    type Error = &'static str;
+    fn try_from(s: &str) -> Result<Self, &'static str> {
+        if s.is_empty() {
+            return Ok(Index::Key(InternalString::default()));
+        }
+
+        let c = s.chars().next().unwrap();
+        if c.is_ascii_digit() {
+            if let Ok(seq) = s.parse::<usize>() {
+                Ok(Index::Seq(seq))
+            } else {
+                let Some(index) = s.find('@') else {
+                    return Err("Invalid index format.");
+                };
+                let (counter, peer) = s.split_at(index);
+                let peer = peer[1..]
```

Comment:
@zxch3n I have fixed the parsing of the tree node index.  But I think `[1..]` is not the best solution.